front/lib/models.ts

```diff
@@ -880,6 +880,54 @@ ChatRetrievedDocument.init(
 
 ChatMessage.hasMany(ChatRetrievedDocument);
 
+export class GensTemplate extends Model<
+  InferAttributes<GensTemplate>,
+  InferCreationAttributes<GensTemplate>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+  declare instructions: string[];
+  declare name: string;
+}
+
+GensTemplate.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    instructions: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+    },
+    name: {
```

Comment:
Let's add a visibility field that express if this is a user template or a workspace template `visibility: "user" | "workspace"` 

Then it becomes easy to have an option to make template workspace wide for builders only